mvrx/src/main/kotlin/com/airbnb/mvrx/Memoize.kt

```diff
@@ -0,0 +1,24 @@
+package com.airbnb.mvrx
+
+import io.reactivex.disposables.Disposable
+
+class Memoize<T>(private val factory: (key: String) -> Pair<T, Disposable>) {
```

Comment:
If we want to move forward with this approach, I'm going to add a bunch of docs and tests. I'm not fully convinced that this is the right path forward yet though since it introduces a number of complexities in the API an handling manual subscriptions.